src/Core/Domain/Carrier/Command/AddCarrierCommand.php

```diff
@@ -41,7 +41,12 @@ public function __construct(
         private string $trackingUrl,
         private int $position,
         private bool $active,
-        private ?string $logoPathName,
+        private bool $hasAdditionalHandlingFee,
+        private bool $isFree,
+        private int $shippingMethod,
+        private int $idTaxRuleGroup,
+        private int $rangeBehavior,
```

Comment:
I think you could introduce a ValueObject for both shippingMehotd and rangeBehavior

Like you did with `OutOfRangeBehaviour`, the value object can validate that the int value is part of a subset, and each allowed value is stored in a const that represents its value like: `OutOfRangeBehaviour::USE_HIGHEST_RANGE`

You can have a similar `ShppingMethod` VO with const values for:
```php
    public const SHIPPING_METHOD_DEFAULT = 0;
    public const SHIPPING_METHOD_WEIGHT = 1;
    public const SHIPPING_METHOD_PRICE = 2;
    public const SHIPPING_METHOD_FREE = 3;
```

Then when you create an add or edit command the input int should be transformed into a VO:
```php
$this->outOfRangerBehaviour = new OutOfRangeBehaviour($rangeBehavior);
```

This will trigger early exceptions with invalid values, which is the workflow used in many other CQRS commands, here's an example:
https://github.com/PrestaShop/PrestaShop/blob/eff832703974bcef13b7ab7ad168a36c7348a48c/src/Core/Domain/Product/Command/UpdateProductCommand.php#L606

This way you don't need to have a public method that is then reused by the validator (like `OutOfRangeBehaviour::assertValue` because the VO validate their content internally

That's the advantage of ValueObject, they are self validating object that prevents using invalid values very early in the process, they will trigger an exception even before the handler or the validator is called, this once you have an instance of a ValueObject you already it can only contain valid value So they are very adapted to this kind of use case with options limited to a subset of values

Reply:
Note: our convention is to use VO in commands only, the QueryResult should still return scalar values (so integers in this case)